flight/Modules/Battery/battery.c

```diff
@@ -149,14 +155,21 @@ static void batteryTask(void * parameters)
 					scaled_voltage = 0.0f;
 					adc_offset_invalid = true;
 				}
+				else if(scaled_voltage > 3.0f && !cells_calculated && (batterySettings.MaxCellVoltage > 0)){
+					cells_calculated = true;
+					cells = ((scaled_voltage / batterySettings.MaxCellVoltage) + 0.9f);
+					flightBatteryData.DetectedCellCount = cells;
+				}
+				else if(!cells_calculated)
+					cells = 1;
 			}
 
 			flightBatteryData.Voltage = scaled_voltage;
 
 			// generate alarms and warnings
-			if (flightBatteryData.Voltage < batterySettings.VoltageThresholds[FLIGHTBATTERYSETTINGS_VOLTAGETHRESHOLDS_ALARM])
+			if (flightBatteryData.Voltage < (batterySettings.VoltageThresholds[FLIGHTBATTERYSETTINGS_VOLTAGETHRESHOLDS_ALARM] * cells))
```

Comment:
Yes, renaming the fields sounds good.

I see now there's a way to turn it off, and that it defaults on.

I am a little concerned that there is the cellcount specified in the UI, which is by default ignored.  It seems like it would be good to use the UI cellcount if specified, and if the UI cellcount is 0 use autodetect.